### PR TITLE
VecGeom single-precision proposal

### PR DESCRIPTION
**Pull request by agheata**
Added VecGeom proposal and removed the `John Doe` entry in the mentor list.

### CODE REVIEW

File: gsoc/2021/mentors.md [hunk 1]
```diff
@@ -3,12 +3,14 @@ title: Summary of Mentors GSoC 2021
 layout: plain
 ---
 
-## Full Mentor List (Name, Email, Org)
+**Note for contributors:** entries must be sorted in last name alphabetic order
```
Comment (xvallspl) [inline gsoc/2021/mentors.md:6]: Shouldn't this go in its own commit PR?

**Review by xvallspl — APPROVED** [review]

File: gsoc/2021/mentors.md (hunk 1 above)
Comment (agheata) [reply gsoc/2021/mentors.md:6]: Yes in principle, but given that it is a minor detail changing the same file, I would let it like this for convenience.

**Comment by agheata** [comment]
@graeme-a-stewart the link checker seems to have problem swallowing https://geant4.web.cern.ch is it because of https?
